### PR TITLE
dev-python/pycuda: Bump version to 2018.1.1

### PR DESCRIPTION
**Pull request by jvesely**
Signed-off-by: Jan Vesely <jano.vesely@gmail.com>

### CODE REVIEW

**Comment by jvesely** [comment]
ping.
This version works with the latest cuda-toolkit.

File: dev-python/pycuda/pycuda-2018.1.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Copyright 1999-2018 Gentoo Foundation
```
Comment (juippis) [inline dev-python/pycuda/pycuda-2018.1.1.ebuild]: This copyright mark is old, you should use the latest repoman to commit to fix it automatically.

File: dev-python/pycuda/pycuda-2018.1.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (juippis) [inline dev-python/pycuda/pycuda-2018.1.1.ebuild]: Can 7 be used in here?

File: dev-python/pycuda/pycuda-2018.1.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
```
Comment (juippis) [inline dev-python/pycuda/pycuda-2018.1.1.ebuild]: Could you try with python3.7?

File: dev-python/pycuda/pycuda-2018.1.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+		-i pycuda/compiler.py || die
+
+	touch siteconf.py || die
+	distutils-r1_python_prepare_all
+}
+
+python_configure() {
+	mkdir -p "${BUILD_DIR}" || die
+	cd "${BUILD_DIR}" || die
+	rm -f ./siteconf.py || die
```
Comment (juippis) [inline dev-python/pycuda/pycuda-2018.1.1.ebuild]: Why `-f`? Is there some condition behind this file?

File: dev-python/pycuda/pycuda-2018.1.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+		--no-use-shipped-boost \
+		$(usex opengl --cuda-enable-gl "") || die
+}
+
+src_test() {
+	# we need write access to this to run the tests
+	addwrite /dev/nvidia0
+	addwrite /dev/nvidiactl
+	python_test() {
+		py.test --debug -v -v -v || die "Tests fail with ${EPYTHON}"
```
Comment (juippis) [inline dev-python/pycuda/pycuda-2018.1.1.ebuild]: I **think** this is nowadays `pytest -vv`

File: dev-python/pycuda/pycuda-2018.1.1.ebuild (hunk 1 above)
Comment (jvesely) [reply dev-python/pycuda/pycuda-2018.1.1.ebuild]: done

File: dev-python/pycuda/pycuda-2018.1.1.ebuild (hunk 2 above)
Comment (jvesely) [reply dev-python/pycuda/pycuda-2018.1.1.ebuild]: done

File: dev-python/pycuda/pycuda-2018.1.1.ebuild (hunk 5 above)
Comment (jvesely) [reply dev-python/pycuda/pycuda-2018.1.1.ebuild]: done

File: dev-python/pycuda/pycuda-2018.1.1.ebuild (hunk 3 above)
Comment (jvesely) [reply dev-python/pycuda/pycuda-2018.1.1.ebuild]: I currently don't have a machine that uses python3.7, but upstream added several fixes for python3.7 in 2018.1.1 so I guess it should work.

File: dev-python/pycuda/pycuda-2018.1.1.ebuild (hunk 4 above)
Comment (jvesely) [reply dev-python/pycuda/pycuda-2018.1.1.ebuild]: That line looks bogus. Probably an overabundance of caution.
There really shouldn't by any file in a newly created directory.

File: dev-python/pycuda/pycuda-2018.1.1.ebuild (hunk 5 above)
Comment (jvesely) [reply dev-python/pycuda/pycuda-2018.1.1.ebuild]: Note that all tests fail in pycuda init on my machine, but I'm not familiar enough with portage sandbox to investigate.
The installed package works fine.

File: dev-python/pycuda/pycuda-2018.1.1.ebuild (hunk 3 above)
Comment (jvesely) [reply dev-python/pycuda/pycuda-2018.1.1.ebuild]: turns out it needs python3.7 enabled for boost which is not available yet.

**Comment by jvesely** [comment]
The package works with dev-util/nvidia-cuda-toolkit-10.0.130 as well.

**Comment by jvesely** [comment]
ping. any more comments on this?

**Comment by jvesely** [comment]
@juippis, is there anything else stopping this?

**Comment by juippis** [comment]
Hey, 

you need someone from the science project to commit it for you. @amadio, @Amynka, @tamiko any takers?

I can't test it since I don't have an nvidia card (running tests required me to recompile my kernel) but it **looks** good to me. You just need a bit more patience for someone from science project to take a look ;)

**Comment by jvesely** [comment]
> Hey,
> 
> you need someone from the science project to commit it for you. @amadio, @Amynka, @tamiko any takers?
> 
> I can't test it since I don't have an nvidia card (running tests required me to recompile my kernel) but it **looks** good to me. You just need a bit more patience for someone from science project to take a look ;)

thanks. do you have any hints on how to ping any of them (@amadio, @Amynka, @tamiko)?

**Comment by amadio** [comment]
@jvesely Sorry for the late reply, I will test and commit.

**Pull request closed without merging**